docs/modules/ROOT/pages/how-tos/configuring/customizing-the-build.adoc

```diff
@@ -239,6 +336,90 @@ The task update PR will trigger the on-pull-request pipeline, testing the change
 
+== Adding sidecar containers to the build pipeline
+
+You can add sidecar containers to the build pipeline to run additional tasks alongside the main container. For example, you can add a sidecar container to run a database to enable your unit tests to run against a real database.
+
+In the example, we add a sidecar container to task to run a PostgreSQL database. Such sidecar can be used to test database seeding, migrations, etc.
+
+[source,yaml]
+--
+spec:
+  tasks:
+  # ...
+   - name: test-db-seeding
+      params:
+        # such task is most likely using source code and will need a trusted artifact to not break the EC.
+        - name: SOURCE_ARTIFACT
+          value: $(tasks.clone-repository-oci-ta.results.SOURCE_ARTIFACT)
+        # database connection parameters, depend on the chosen image
+        - name: PGSQL_USER
+          value: "jdoe"
+        - name: PGSQL_PASSWORD
+          value: "foobar"
+        - name: PGSQL_HOSTNAME
+          value: "0.0.0.0"
+        - name: PGSQL_PORT
+          value: "5432"
+        - name: PGSQL_DATABASE
+          value: "db"
+      runAfter:
+        - clone-repository-oci-ta
+      taskSpec:
+        params:
+          - description: The Trusted Artifact URI pointing to the artifact with the application source code.
+            name: SOURCE_ARTIFACT
+            type: string
+        volumes:
+          - name: workdir
+            emptyDir: {}
+        stepTemplate:
+          volumeMounts:
+            - mountPath: /var/workdir
+              name: workdir
+        # define the sidecar containers
+        sidecars:
+        - image: quay.io/sclorg/postgresql-15-c9s:latest
+          name: seeding-test-db
+          env:
+            - name: POSTGRESQL_DATABASE
+              value: $(params.PGSQL_DATABASE)
+            - name: POSTGRESQL_USER
+              value: $(params.PGSQL_USER)
+            - name: POSTGRESQL_PASSWORD
+              value: $(params.PGSQL_PASSWORD)
+          # Configure the env variables required for tests to connect to the database. The variables are specific to the image/database used.
+          env:
+            - name: DATABASE_URL
+              value: "postgresql://$(params.PGSQL_USER):$(params.PGSQL_PASSWORD)@localhost:$(params.PGSQL_PORT)/$(params.PGSQL_DATABASE)"
+            - name: DATABASE_SSL
+              value: "false"
+        steps:
+          - name: use-trusted-artifact
+            image: quay.io/redhat-appstudio/build-trusted-artifacts:latest@sha256:8391272c4e5011120e9e7fee2c1f339e9405366110bf239dadcbc21e953ce099
+            args:
+              - use
+              - $(params.SOURCE_ARTIFACT)=/var/workdir
```

Comment:
I guess that this is just needed if your sidecar content is pulled from the git source? This code snippet isn't in the TA section so we need to be careful about the actual content we include without explaining it further.

Reply:
It is yeah. The main purpose of these examples is to show how to extend pipelines with custom tasks and comply with the EC. But I can see your point.  Maybe instead of these full examples, it should be made very obvious that pipeline customizations need a different approach, especially if the builds are using "external" policies not defined by the project itself.